src/functionalTest/java/uk/gov/hmcts/dts/fact/admin/AdminCourtContactEndpointTest.java

```diff
@@ -0,0 +1,243 @@
+import uk.gov.hmcts.dts.fact.model.admin.Contact;
+import uk.gov.hmcts.dts.fact.model.admin.ContactType;
+import uk.gov.hmcts.dts.fact.util.AdminFunctionalTestBase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpHeaders.*;
+import static org.springframework.http.HttpStatus.*;
+import static uk.gov.hmcts.dts.fact.util.TestUtil.*;
+
+@ExtendWith(SpringExtension.class)
+public class AdminCourtContactEndpointTest extends AdminFunctionalTestBase {
+    private static final String CONTACTS_PATH = "/" + "contacts";
+    private static final String CONTACT_TYPES_PATH = "contactTypes";
+    private static final String BIRMINGHAM_CIVIL_AND_FAMILY_JUSTICE_CENTRE_SLUG = "birmingham-civil-and-family-justice-centre";
+    private static final int ENQUIRIES_CONTACT_TYPE_ID = 22;
+    private static final String ENQUIRIES_CONTACT_TYPE = "Enquiries";
+    private static final String ENQUIRIES_CONTACT_TYPE_WELSH = "Ymholiadau";
+    private static final String TEST_NUMBER = "test number";
+    private static final Contact TEST_CONTACT = new Contact(ENQUIRIES_CONTACT_TYPE_ID, TEST_NUMBER, "explanation", "explanation cy");
+
+    @Test
+    public void shouldRetrieveContacts() {
+        final var response = given()
+            .relaxedHTTPSValidation()
+            .header(CONTENT_TYPE, CONTENT_TYPE_VALUE)
+            .header(AUTHORIZATION, BEARER + authenticatedToken)
+            .when()
+            .get(ADMIN_COURTS_ENDPOINT + BIRMINGHAM_CIVIL_AND_FAMILY_JUSTICE_CENTRE_SLUG + CONTACTS_PATH)
+            .thenReturn();
+
+        assertThat(response.statusCode()).isEqualTo(OK.value());
+        final List<Contact> contacts = response.body().jsonPath().getList(".", Contact.class);
+        assertThat(contacts).hasSizeGreaterThan(1);
+    }
+
+    @Test
+    public void shouldRequireATokenWhenRetrievingContacts() {
+        final var response = given()
+            .relaxedHTTPSValidation()
+            .header(CONTENT_TYPE, CONTENT_TYPE_VALUE)
+            .when()
+            .get(ADMIN_COURTS_ENDPOINT + BIRMINGHAM_CIVIL_AND_FAMILY_JUSTICE_CENTRE_SLUG + CONTACTS_PATH)
+            .thenReturn();
+
+        assertThat(response.statusCode()).isEqualTo(UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void shouldBeForbiddenForRetrievingContacts() {
+        final var response = given()
+            .relaxedHTTPSValidation()
+            .header(CONTENT_TYPE, CONTENT_TYPE_VALUE)
+            .header(AUTHORIZATION, BEARER + forbiddenToken)
+            .when()
+            .get(ADMIN_COURTS_ENDPOINT + BIRMINGHAM_CIVIL_AND_FAMILY_JUSTICE_CENTRE_SLUG + CONTACTS_PATH)
+            .thenReturn();
+
+        assertThat(response.statusCode()).isEqualTo(FORBIDDEN.value());
+    }
+
+    @Test
+    public void shouldAddOrRemoveContacts() throws JsonProcessingException {
```

Comment:
Looks a bit long to me! Is there a way to break this into two tests, add and remove? 

I imagine that the purpose of it was to do both in one like a setup/teardown :D which is fine if so. Just thinking from a de bugging point of view. Your call!

Reply:
Think I am going to leave both add and remove in the same test so the database is back to its original state after the test, but I am going to move that bit where it verify the results using the existing 'courts' endpoint out to a private method to make this method looks a bit smaller.